lib/Basic/LangOptions.cpp

```diff
@@ -564,10 +568,14 @@ std::pair<bool, bool> LangOptions::setTarget(llvm::Triple triple) {
   }
 
   // Set the "_pointerBitWidth" platform condition.
-  if (Target.isArch32Bit()) {
-    addPlatformConditionValue(PlatformConditionKind::PointerBitWidth, "_32");
-  } else if (Target.isArch64Bit()) {
-    addPlatformConditionValue(PlatformConditionKind::PointerBitWidth, "_64");
+  if (Target.getArch()==llvm::Triple::ArchType::avr) {
+    addPlatformConditionValue(PlatformConditionKind::PointerBitWidth, "_16");
```

Comment:
Can this be `if (Target.isArch16Bit()) {…}` instead?
(To support not only `avr`, but also `msp430`, etc.)

Would also need to update [SupportedConditionalCompilationPointerBitWidths](https://github.com/swiftlang/swift/blob/swift-5.10.1-RELEASE/lib/Basic/LangOptions.cpp#L89-L92) array.

Reply:
good idea.

I'm also thinking now, although it's slightly pedantic, this change and the other you suggested, would be better in https://github.com/swiftlang/swift/pull/74693/files which is the PR I raised for adding/improving 16-bit pointer support... separate from this PR which is about AVR only.

Reply:
OK done. This change is now written as you suggested, with [SupportedConditionalCompilationPointerBitWidths](https://github.com/swiftlang/swift/blob/swift-5.10.1-RELEASE/lib/Basic/LangOptions.cpp#L89-L92) too. Both changes moved to https://github.com/swiftlang/swift/pull/74693.